examples/basic-datasource/pkg/query/scenario/frame.go

```diff
@@ -0,0 +1,17 @@
+package scenario
+
+import (
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+)
+
+func NewDataFrame(query backend.DataQuery) *data.Frame {
+	switch query.QueryType {
```

Comment:
This is great! Didn't know about this. Will update to use this instead 🙏🏻

Reply:
I tried this out but I basically needs to repeat a lot of code and it feels a bit weird to route based on queryType. It would make sense if we where expecting different incoming models but that isn't the case in the frontend.

Reply:
Okay make sense